hack/tools/conversion-verifier/main.go

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"go/types"
+	"path"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+	crdmarkers "sigs.k8s.io/controller-tools/pkg/crd/markers"
+	"sigs.k8s.io/controller-tools/pkg/loader"
+	"sigs.k8s.io/controller-tools/pkg/markers"
+)
+
+var (
+	storageVersion   = markers.Must(markers.MakeDefinition("kubebuilder:storageversion", markers.DescribesType, crdmarkers.StorageVersion{}))
+	nolintConversion = markers.Must(markers.MakeDefinition("kubebuilder:conversion:nolint", markers.DescribesType, false))
+
+	apiPackages         = map[string]map[string]*loader.Package{}
+	storageVersionTypes = map[string]*storageVersionType{}
+)
+
+type versionType struct {
+	base string
```

Comment:
Can we add a few godoc comments around those types? I understood it when I stepped through the code, but it's hard with only reading the code and not being familiar with the controller-tools APIs we're using